File: Data Structures & Algorithms/binary-search/submission-4.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.b_search(0,len(nums)-1,nums,target)

    def b_search(self,l,r,nums,target):
        if l>r:
            return -1
        m=l+(r-l)//2
        if nums[m]==target:
            return m
        if nums[m]<target:
            return self.b_search(m+1,r,nums,target)
        else:
            return self.b_search(l,m-1,nums,target)
        


    
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const res={}
        for(let num of nums){
            res[num]=(res[num]||0)+1
        }
        let arr=Object.entries(res).map(([num,freq])=>[freq,num])
        arr.sort((a,b)=>b[0]-a[0])

        return arr.slice(0,k).map((pair)=>pair[1])
    }
}
